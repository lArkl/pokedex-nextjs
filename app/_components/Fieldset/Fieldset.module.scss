.container {
  border: none;
  margin: 0;
  padding: 0;
}

.text {
  text-align: start;
  margin-bottom: 0.5rem;
}
