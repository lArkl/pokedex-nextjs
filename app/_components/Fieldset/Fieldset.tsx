import { FC, ReactNode } from 'react'
import Typography from '../Typography'
import classNames from 'classnames'
import styles from './Fieldset.module.scss'

export interface FieldsetProps {
  label?: string
  name: string
  className?: string
  textClassName?: string
  children: ReactNode
}

const Fieldset: FC<FieldsetProps> = ({ label, name, className, textClassName, children }) => {
  return (
    <fieldset className={classNames(styles.container, className)}>
      <label htmlFor={name}>
        <Typography className={classNames(styles.text, textClassName)} variant="md">
          {label ?? name}
        </Typography>
      </label>
      {children}
    </fieldset>
  )
}

export default Fieldset
