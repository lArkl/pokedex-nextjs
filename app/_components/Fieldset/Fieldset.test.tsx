import { render, screen } from '@testing-library/react'
import Fieldset from './Fieldset'

describe('Fieldset', () => {
  it('renders component', async () => {
    const { asFragment } = render(<Fieldset name="field">Form field set</Fieldset>)

    expect(asFragment()).toMatchSnapshot()
  })
  it('shows label', async () => {
    render(
      <Fieldset name="field">
        <input id="field" />
      </Fieldset>,
    )

    expect(screen.getByRole('group')).toBeInTheDocument()
    expect(screen.getByLabelText(/field/)).toBeInTheDocument()
  })

  it('accepts a className', async () => {
    render(
      <Fieldset className="class" name="field">
        form value
      </Fieldset>,
    )

    expect(screen.getByRole('group')).toHaveClass('class')
  })
})
