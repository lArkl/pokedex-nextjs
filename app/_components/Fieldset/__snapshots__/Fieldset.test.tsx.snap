// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fieldset > renders component 1`] = `
<DocumentFragment>
  <fieldset
    class="_container_3e604f"
  >
    <label
      for="field"
    >
      <p
        class="_md_bcfef7 _text_3e604f"
      >
        field
      </p>
    </label>
    Form field set
  </fieldset>
</DocumentFragment>
`;
